Add random structural diary plugin

ShockThunder opened the pull request:
# I am submitting a new Community Plugin
## Repo URL
Link to my plugin: https://github.com/ShockThunder/RandomStructuralDiary

Release Checklist
 - [x] I have tested this on Windows, macOS, and Linux (if applicable)
 - [x] My GitHub release contains all required files
   - [x] main.js
   - [x] manifest.json
   - [ ] styles.css (optional)
 - [x] GitHub release name matches the exact version number specified in my manifest.json (Note: Use the exact version number, don't include a prefix v)
 - [x] The id in my manifest.json matches the id in the community-plugins.json file.
 - [x] README clearly describes the plugins purpose and provides clear usage instructions.
 - [x] I have added a license in the LICENSE file.

lishid commented:
https://github.com/ShockThunder/RandomStructuralDiary/blob/b4fc1f319474a8a75af11b2b5af7b2a2b4993ddc/main.ts#L17

Please rename your classes/interfaces from the sample names to your plugin's actual name.

https://github.com/ShockThunder/RandomStructuralDiary/blob/b4fc1f319474a8a75af11b2b5af7b2a2b4993ddc/main.ts#L31

Don't do this - instead, use `vault.getAbstractFileByPath(path)` and then check for `file instanceof TFile`. Then you can use `vault.cachedRead` rather than `adapter.read()`. You should also use `async` and `await` instead of `.then` when possible.

Here I don't think you need a `checkCallback` either - you can just convert it into a regular `callback` command.

https://github.com/ShockThunder/RandomStructuralDiary/blob/b4fc1f319474a8a75af11b2b5af7b2a2b4993ddc/main.ts#L84
Make sure you test for null here and check that it's actually a markdown file. You are also completely overwriting this file's content, which may cause data-loss if the user was working on something.

ShockThunder commented:
Thank you for your review.
I updated the code.
I decided not to create a default file and use one in the plugin folder.

Unfortunately, i couldn't get file in plugin folder by `vault.getAbstractFileByPath(path)`, so, i use the adapter
https://github.com/ShockThunder/RandomStructuralDiary/blob/a0ca0f1c6bc06bd8f105d43db691ba1cc45c8044/main.ts#L37

I changed `checkCallback` to async `callback` and remove `then` in promises.

Also, I save user content and place my generated text after this.

lishid commented:
> Unfortunately, i couldn't get file in plugin folder by vault.getAbstractFileByPath(path), so, i use the adapter
https://github.com/ShockThunder/RandomStructuralDiary/blob/a0ca0f1c6bc06bd8f105d43db691ba1cc45c8044/main.ts#L37

Sure, files in the config folder (including the plugin folder) doesn't exist in the vault structure, so using the adapter is correct.
